# Raw data to variant calls

Processing Illumina reads to raw vcf file

Data

### 1. ddRAD

6 Illumina HiSeq 2000 libraries. (4 libraries on HD - Elements. 2 on FGCZ server)

48 individuals pooled per library.

9 populations - pops sequenced across at least 4 lanes each. Two additional populations (TP & GTT) were sequenced with this project, but are not part of the analyses.

Paired-end ddRAD; 2 files per library

Barcodes and indiv names in Samples_Barcodes.txt

Restriction enzymes

```
PstI TGCAG - 3'

EcoRI AATT - 3'
```


### 2. WGS

32 individuals sequenced by Edinburgh Genomics using NBAF grant to at least 15X

TruSeq PCR-free library

1 lane 150bp PE HiSeq X: 370M x 370M reads




## Demultiplex and adapter filtering

This has been done previously for ddRAD data [here](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/1.RawData_to_Variants.md)

WGS data has been demultiplexed by Edinburgh Genomics



## Mapping to Reference genome

We have access to the pre-released chromosome level Sanger genome [here](ftp://ngs.sanger.ac.uk/scratch/project/grit/VGP/ilAriAges1.1/)

Contacts: James Torrance and Mark Blaxter. Email exchange on UCL email 12 Feb 2021. 

Prepare a list of samples to be mapped: 
```
pwd
/newhome/aj18951/1a_Aricia_agestis_GWASdata

#list of the ddRAD raw fastq files
ls ../1a_Aricia_agestis_PopGenomics/demultiplexed.ipyrad_lib1.6/*R1*gz > R1.names
ls ../1a_Aricia_agestis_PopGenomics/demultiplexed.ipyrad_lib1.6/*R1*gz > R2.names

#substitute path to have only the sample names for R1 and R2 
sed -i 's:../1a_Aricia_agestis_PopGenomics/demultiplexed.ipyrad_lib1.6/::' *names

#split into lists of 100 - the max sample number we can submit to the cluster
split -l 100 R*.names 
```

Mapped ddRAD data to the reference genome using this [script](https://github.com/alexjvr1/AriciaAgestis_GWASMS/blob/main/02_MapwithBWAmem.ARRAY1.sh)

Make enough array scripts to run all the samples (ARRAY1-6.sh)


## Call variants 

Call varaints using samtools mpileup and bcftools call 

Create the output folder
```
mkdir 03_variants

```

Use [this](https://github.com/alexjvr1/AriciaAgestis_GWASMS/blob/main/03a_variant_calling_bluecp3.sh) script to create the variant calling submission script based on the mapped reads. 

