# Filtering raw variants

## ddRAD data


Filtering variants from the raw vcf output to produce one dataset:

1. GWAS analysis using BSLMM (include as many loci as possible).


### Filters to apply

1. Remove loci with QUAL < 100 (i.e. Phred confidence in variant site)

2. Minimum mean depth of 6 (i.e. remove loci with lower than mean 6x depth)

3. Max mean depth of mean + 2xSD of meanDP (here 500)

4. remove all multi-allelic SNPs

5. Remove all loci genotyped in <30% of individuals

6. Remove individuals with >60% missingness



The initial variants.raw.bcf file 
```
#Concatenate all the individual bcf files that have been called (one for each chromosome)
module load apps/bcftools-1.8

ls *raw.bcf > FILE
bcftools concat -O b -a -d none -f FILE > Aagestis.raw.bcf


#The raw bcf files need to be processed to "see" missing data

bcftools filter -S . -O u -e 'FMT/DP=0' Aagestis.raw.bcf |bcftools view -O b -o Aagestis.withmissing.bcf

#If you want to run this per chromosome.. 
#for i in $(ls *bcf); do bcftools filter -S . -O u -e 'FMT/DP=0' $i |bcftools view -O b -o $i.withmissing.bcf; done

#You can view the bcf file with bcftools. Check that the 0 depth genotypes have been called as missing (./.) and not REF (0/0)
bcftools view Aagestis.withmissing.bcf |less

#convert the bcf to vcf
bcftools convert -O v Aagestis.withmissing.bcf -o Aagestis.withmissing.vcf
```


Now we can start filtering. First filter on depth 
```
Dataset1

#We're using vcftools to filter the dataset
module load apps/vcftools-0.1.12b

#First get a list of all the sample names. There are two populations (GTT and TP) that we won't use for the GWAS so they need to be removed. 

bcftools query -l Aagestis.withmissing.vcf > samplenames
grep GTT samplenames > toremove
GTT_1_R1_.fastq.gz
GTT_3_R1_.fastq.gz
GTT_4_R1_.fastq.gz
GTT_5_R1_.fastq.gz
GTT_6_R1_.fastq.gz

grep TP samplenames >> toremove
TP_31_R1_.fastq.gz
TP_32_R1_.fastq.gz
TP_33_R1_.fastq.gz
TP_43_R1_.fastq.gz
TP_44_R1_.fastq.gz

vcftools --vcf Aagestis.withmissing.vcf --remove toremove --recode --recode-INFO-all --out AAgestis.withmissing.276_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf Aagestis.withmissing.vcf
	--exclude toremove
	--recode-INFO-all
	--out AAgestis.withmissing.276_FINAL
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 276 out of 286 Individuals
Outputting VCF file...
After filtering, kept 558840 out of a possible 558840 Sites
Run Time = 342.00 seconds


#implement some initial basic filters 
vcftools --vcf AAgestis.withmissing.276_FINAL.recode.vcf --minQ 20 --min-meanDP 6 --max-alleles 2 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.withmissing.276_FINAL.recode.vcf
	--recode-INFO-all
	--max-alleles 2
	--min-meanDP 6
	--minQ 20
	--max-missing 0.3
	--out AAgestis_FINAL
	--recode

After filtering, kept 276 out of 276 Individuals
Outputting VCF file...
After filtering, kept 135150 out of a possible 558840 Sites
Run Time = 127.00 seconds



#We need to find the max Depth filter. We have this because some loci might've incorrectly been lumped together (e.g gene duplicates or very similar genetic regions). Typically these will have ~2x the coverage of the other loci. So we need to find the mean depth across all loci and then find the cut-off

vcftools --vcf AAgestis_FINAL.recode.vcf --site-mean-depth

awk '!/IN/' out.ldepth.mean | cut -f3 > lmeandepth

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
binwidth=0.1
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'lmeandepth' using (bin( $1,binwidth)):(1000.0) smooth freq with boxes
pause -1
EOF

vcftools --vcf AAgestis_FINAL.recode.vcf --minQ 20 --min-meanDP 6 --max-meanDP 500 --max-alleles 2 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis_FINAL2

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis_FINAL.recode.vcf
	--recode-INFO-all
	--max-alleles 2
	--max-meanDP 500
	--min-meanDP 6
	--minQ 20
	--max-missing 0.3
	--out AAgestis_FINAL2
	--recode

After filtering, kept 276 out of 276 Individuals
Outputting VCF file...
After filtering, kept 122503 out of a possible 135150 Sites
Run Time = 92.00 seconds


```



Remove individuals with a high proportion of missingness.

```
vcftools --vcf AAgestis_FINAL2.recode.vcf --missing-indv

awk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```
The vast majority of the samples have <30% missing data, although there are a few with up to 100% missingness


![alt_txt][Fig1]

[Fig1]:https://user-images.githubusercontent.com/12142475/109166384-8ec32f00-7774-11eb-84cb-50452e1dc2d2.png


```
#Remove individuals with >60% missingness

awk '$5>0.6 {print $1}' out.imiss > indivstoremove

cat indivstoremove

INDV
BAR_1_2013_R1_.fastq.gz
BAR_22_2014_R1_.fastq.gz
BAR_23_2014_R1_.fastq.gz
BCH_38_2013_R1_.fastq.gz
BRO_4_2014_R1_.fastq.gz
FOR_22_2014_R1_.fastq.gz
FOR_26_2014_R1_.fastq.gz
HOD_13_2014_R1_.fastq.gz
HOD_18_2014_R1_.fastq.gz
HOD_6_2014_R1_.fastq.gz
HOD_8_2014_R1_.fastq.gz
LYD_20_2014_R1_.fastq.gz
LYD_28_2014_R1_.fastq.gz
MOF_42_2014_R1_.fastq.gz
SWD_17_2013_R1_.fastq.gz
```


Remove samples

```
vcftools --vcf AAgestis_FINAL2.recode.vcf --remove indivstoremove --recode --recode-INFO-all --out AAgestis.261_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis_FINAL2.recode.vcf
	--exclude indivstoremove
	--recode-INFO-all
	--out AAgestis.261_FINAL
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 261 out of 276 Individuals
Outputting VCF file...
After filtering, kept 122503 out of a possible 122503 Sites
Run Time = 75.00 seconds


#And check if the filters remove any more loci:

vcftools --vcf AAgestis.261_FINAL.recode.vcf --minQ 20 --min-meanDP 6 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis.261_FINAL.recode.vcf

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.261_FINAL.recode.vcf
	--recode-INFO-all
	--min-meanDP 6
	--minQ 20
	--max-missing 0.3
	--out AAgestis.261_FINAL.recode.vcf
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 122428 out of a possible 122503 Sites
Run Time = 87.00 seconds


```

Rename the individuals in this file
```
module load apps/bcftools-1.8

bcftools query -l AAgestis.261_FINAL.recode.vcf > A261.oldnames

sed 's/_R1_.fastq.gz//g' A261.oldnames > A261.newnames

bcftools reheader AAgestis.261_FINAL.recode.vcf -s A261.newnames -o AAgestis.261_FINAL.newnames.vcf
```


### Assess the data

These stats are based on the dataset before filtering for MAF 1% (122428 loci)

You can use vcftools for this: 
```
module load apps/vcftools-0.1.17.2
```

OUTPUT DEPTH STATISTICS

--depth

Generates a file containing the mean depth per individual. This file has the suffix ".idepth".

--site-depth

Generates a file containing the depth per site summed across all individuals. This output file has the suffix ".ldepth".

--site-mean-depth

Generates a file containing the mean depth per site averaged across all individuals. This output file has the suffix ".ldepth.mean".


**NOTE

If there is a comma in a descriptor in the vcf header, you might get an error with the latest versions of vcftools.
```
Warning: Expected at least 2 parts in INFO entry: ID=PV4,Number=4,Type=Float,Description="P-values for strand bias, baseQ bias, mapQ bias and tail distance bias">
Warning: Expected at least 2 parts in INFO entry: ID=PV4,Number=4,Type=Float,Description="P-values for strand bias, baseQ bias, mapQ bias and tail distance bias">
Warning: Expected at least 2 parts in INFO entry: ID=DP4,Number=4,Type=Integer,Description="Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases">
Warning: Expected at least 2 parts in INFO entry: ID=DP4,Number=4,Type=Integer,Description="Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases">
```
Simply remove the comma in the Description="" part of the line. Or ignore the warning. See [here](https://github.com/vcftools/vcftools/issues/134) for validation. 


#### 1. Mean depth per site per population

```
/newhome/aj18951/1a_Aricia_agestis_GWASdata/03_variants

vcftools --vcf AAgestis.261_FINAL.newnames.vcf --site-depth

```

Plot
```
Mean depth per site in raw & filtered data
DP.sites <- read.table("out.ldepth", header=T)

pdf("a1_AriciaAgestis_DepthPerSite.pdf")
ggplot(DP.sites, aes(SUM_DEPTH)) + geom_histogram()
dev.off()


summary(DP.sites.filtered)
         CHROM            POS           SUM_DEPTH      SUMSQ_DEPTH      
 contig_365  :   82   Min.   :     4   Min.   : 2518   Min.   :   59081  
 m_scaff_3898:   80   1st Qu.:  2275   1st Qu.:13229   1st Qu.: 1643958  
 m_scaff_714 :   80   Median :  5135   Median :29046   Median : 5223654  
 contig_2942 :   79   Mean   :  7790   Mean   :30056   Mean   : 6068789  
 m_scaff_5830:   78   3rd Qu.: 10551   3rd Qu.:45418   3rd Qu.: 9723112  
 contig_16698:   77   Max.   :147561   Max.   :68598   Max.   :18003826  
 (Other)     :62578                                                      
   dataset         
 Length:63054      
 Class :character  
 Mode  :character  
                   
                   
summary(DP.sites)
          CHROM             POS           SUM_DEPTH      SUMSQ_DEPTH      
 m_scaff_6129:   176   Min.   :     4   Min.   :    1   Min.   :       1  
 m_scaff_5252:   168   1st Qu.:  2197   1st Qu.:    4   1st Qu.:       9  
 m_scaff_6232:   153   Median :  5058   Median :   39   Median :     226  
 contig_2942 :   152   Mean   :  7780   Mean   : 7724   Mean   : 1528150  
 m_scaff_910 :   150   3rd Qu.: 10497   3rd Qu.: 4038   3rd Qu.:  390659  
 m_scaff_714 :   144   Max.   :147699   Max.   :71626   Max.   :18744992  
 (Other)     :281149                                                      
```

![alt_txt][Fig3]

[Fig3]:https://user-images.githubusercontent.com/12142475/109176648-ef576980-777e-11eb-95b9-4fc630358020.png



#### 2. Individual mean depth vs number of genotyped loci. Coloured per population.

#### 3. Coverage per individual and per pop






Plot of mean depth per individual grouped by population: 
/newhome/aj18951/1a_Aricia_agestis_GWASdata/03_variants

```
module load languages/R-4.0.3-gcc9.1.0

R version 4.0.3 (2020-10-10) -- "Bunny-Wunnies Freak Out"
Copyright (C) 2020 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

library(ggplot2)

DP.indivs <- read.table("out.idepth", header=T)
head(DP.indivs)
DP.indivs$pop <- gsub("_.*_.*", "", DP.indivs$INDV)
head(DP.indivs)

pdf("1a_AriciaAgestis_MeanDP.pdf")
ggplot(DP.indivs, aes(x=pop, y=MEAN_DEPTH)) + geom_boxplot()
dev.off()
```


![alt_txt][Fig2]

[Fig2]:https://user-images.githubusercontent.com/12142475/109176002-4446b000-777e-11eb-97bb-deb21be10b41.png







Number of variants per population (with individual variation)
```
out.imiss <- read.table("out.imiss", header=T)
out.imiss$N_Variants <- (out.imiss$N_DATA-out.imiss$N_MISS)
out.imiss$pop <- gsub("_.*_.*", "", out.imiss$INDV)

pdf("1a_AriciaAgestis_nrVariantsPerPop.pdf")
ggplot(out.imiss, aes(x=pop, y=N_Variants)) + geom_boxplot()
dev.off()
```
![alt_txt][Fig4]

[Fig4]:https://user-images.githubusercontent.com/12142475/109176940-3b0a1300-777f-11eb-8014-d81b923ee6f8.png

