# Pop Structure and variance within pop

Final Datasets used here: 

1. 1% MAF - for Pop Gen Stats

261 indivs

74738 variants

9207 loci

2. 5% MAF - for outlier analyses

261 indivs

43541 variants

8405 loci

This is > double the loci and variants used in the Pop gen study (mapped to the de novo A.agestis assembly). See [here](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/3.GeneticDiversity%26Structure.md) 

## PopGen 

### Heterozygosity

```
/newhome/aj18951/1a_Aricia_agestis_GWASdata/04_Outlier

module load languages/R-4.0.3-gcc9.1.0

library(adegenet)
library(vcfR)
library(hierfstat)
```





## Outlier Analyses

### PCAdapt

First estimate genetic diversity and pop structure with PCA and then outlier analysis. 

#### NB - there are 4 WIS samples that are outliers in the original (de novo) PCA, which are Extreme outliers in Kristian (MSc student's) PCA based on the Sanger genome data. 

What are these indivs? Is there a problem with their data? 

```
/newhome/aj18951/1a_Aricia_agestis_GWASdata/04_Outlier

module load languages/R-4.0.3-gcc9.1.0

vcftools --vcf AAgestis.261.MAF0.01.FINAL.recode.vcf --max-missing 0.8 --recode --recode-INFO-all --out AAgestis.261.MAF0.01.maxMiss0.8 

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.261.MAF0.01.FINAL.recode.vcf
	--recode-INFO-all
	--max-missing 0.8
	--out AAgestis.261.MAF0.01.maxMiss0.8
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 49430 out of a possible 74738 Sites


module load apps/plink2

vcftools --vcf AAgestis.261.MAF0.01.maxMiss0.8.recode.vcf --plink --out AA261.plink

plink --file AA261.plink --recode A --out AA261.plink

PLINK v1.90b3f 64-bit (2 Mar 2015)         https://www.cog-genomics.org/plink2
(C) 2005-2015 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to AA261.plnk.log.
129152 MB RAM detected; reserving 64576 MB for main workspace.
.ped scan complete (for binary autoconversion).
Performing single-pass .bed write (49430 variants, 261 people).
--file: AA261.plnk-temporary.bed + AA261.plnk-temporary.bim +
AA261.plnk-temporary.fam written.
49430 variants loaded from .bim file.
261 people (0 males, 0 females, 261 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA261.plnk.nosex .
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 261 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.939274.
49430 variants and 261 people pass filters and QC.
Note: No phenotypes present.
--recode A to AA261.plnk.raw ... done.
```

93% genotyping rate

Open R and run analysis
```


```







